### PR TITLE
Add suppression diff rule for time formats

### PR DESCRIPTION
**Pull request by MZochowskiSUMO**
This PR adds suppression rule for "normal" CRUD TF resources. The case for `sumologic_content` and its JSONs was filed as a separate issue: https://github.com/SumoLogic/terraform-provider-sumologic/issues/444

### CODE REVIEW

File: sumologic/util.go
Comment (MZochowskiSUMO) [inline sumologic/util.go:316]: Is this the correct "go" way of handling errors?

File: sumologic/util.go
Comment (MZochowskiSUMO) [inline sumologic/util.go:384]: I had to return sth, so I decided to return 0. Is there a better way to do it?

File: sumologic/util_test.go
Comment (MZochowskiSUMO) [inline sumologic/util_test.go:8]: I added a unit test here.
I am not sure we can create an acceptance test for the case where we want to assert that the plan was empty. If you know how to do it, please let me know, I will write such acceptance test.

File: sumologic/util.go
Comment (sumovishal) [inline sumologic/util.go:305]: A function header listing out examples of time differences we are suppressing would be helpful.

File: sumologic/util_test.go
Comment (sumovishal) [inline sumologic/util_test.go:1]: Unit test is great. Let's update some acceptance tests in `resource_sumologic_dashboard_test.go` to make sure we have coverage in acceptance tests too.

**Review by sumovishal — COMMENTED** [review]
Made first pass, left some comments.

File: sumologic/util_test.go
Comment (MZochowskiSUMO) [reply sumologic/util_test.go:1]: I thought about it, but I didn't find a way to test that the plan is empty. Do you know how to write such a test?

Comment (MZochowskiSUMO) [reply sumologic/util_test.go:1]: I can add a positive/negative variant (whichever lacks) there, but I don't see how it tests this PR, which only changes the emptiness of a plan.

File: sumologic/util.go
Comment (sumovishal) [reply sumologic/util.go:305]: yeah, a comment above the function.

File: sumologic/util_test.go
Comment (sumovishal) [reply sumologic/util_test.go:1]: Yes to what Greg said. 

>I can add a positive/negative variant (whichever lacks) there, but I don't see how it tests this PR

With this PR, there should no diff in the acceptance test if we use one of the variants. Like `-24h` or `-1d` should not show any diff in acceptance test.

Comment (MZochowskiSUMO) [reply sumologic/util_test.go:1]: Hi, I agree that there is a value in such tests. However, as I mentioned I didn't find a way to assert that a test is empty (only that it is not empty). Do you know how to do it/whether it is possible?

Comment (sumovishal) [reply sumologic/util_test.go:1]: >I didn't find a way to assert that a test is empty (only that it is not empty). 

You don't need to assert for anything. Having one of the time range variant like `-24h` in the acceptance test will ensure new code is tested. The backend will return `-1d` in such case, and the custom diff function will make sure there is no diff between state and the tf file.

Comment (MZochowskiSUMO) [reply sumologic/util_test.go:1]: Thanks, I didn't know that the acc test already checks whether the plan was empty (maybe that's why I didn't find any flag to enforce such check :) ). I modified tests for dashboards, but I will also do it for monitors, as my change is applicable also there.

File: sumologic/util.go
Comment (sumovishal) [reply sumologic/util.go:325]: >When I tested it I didn't get any such logs (here and in the other parts of our provider).

You need to prefix terraform commands with `TF_LOG=DEBUG` to see these logs.

File: sumologic/util.go
Comment (sumovishal) [reply sumologic/util.go:333]: yeah, this looks right to me.

**Review by sumovishal — APPROVED** [review]
LGTM, please wait for a review from the Alerting team. cc @rmamilla-sumo @tarunk2

File: sumologic/util.go
Comment (MZochowskiSUMO) [reply sumologic/util.go:325]: thanks!

File: sumologic/util.go
Comment (rmamilla-sumo) [inline sumologic/util.go:315]: @MZochowskiSUMO , Is user allowed to set isRelative field ? why are we allowing it if there is no difference if user enters -ve / +ve value

Comment (MZochowskiSUMO) [reply sumologic/util.go:315]: @rmamilla-sumo No it is not available to users. isRelative is only a flag we use in code (to suppress diffs) to specify if we treat -ve / +ve values as the same value. In most places there are equivalent, but here afaik they are not: https://github.com/SumoLogic/terraform-provider-sumologic/pull/442/files#diff-4bc5e99d3cf4cbda61368cb6fdbafe1bc97884202cca74d578e12b053064cb25R176, https://github.com/SumoLogic/terraform-provider-sumologic/pull/442/files#diff-b749eb8ae5017ef0ed2545e094d7e38321f015a91b335265140abcf75fb61365R503 (relativeTimeRangeSchema). Can you confirm that?

Comment (MZochowskiSUMO) [reply sumologic/util.go:315]: @rmamilla-sumo We are suppressing time diff values treating -ve/+ve as equal everywhere except the 2 places above. isRelative flag only changes the semantics of diff suppression, it does not enable/disable it.

**Review by rmamilla-sumo — APPROVED** [review]
LGTM

**Review by tarunk2 — APPROVED** [review]

File: sumologic/util.go
Comment (MZochowskiSUMO) [reply sumologic/util.go:315]: I changed suppression to absolute (isRelative = false) in SLO monitors' trigger_conditions.

**Pull request merged**
